snapcraft/storeapi/__init__.py

```diff
@@ -13,9 +13,282 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import hashlib
+import json
+import logging
+import os
 
-from .channels import get_channels, update_channels  # noqa
+from urllib import parse
+
+import requests
+from ssoclient import v2 as sso
+
+import snapcraft
+from snapcraft import config
+from .constants import (
+    DEFAULT_SERIES,
+    UBUNTU_SSO_API_ROOT_URL,
+    UBUNTU_STORE_API_ROOT_URL,
+    UBUNTU_STORE_SEARCH_ROOT_URL,
+)
+
+from . import _upload
 from .info import get_info  # noqa
-from ._login import login  # noqa
-from ._download import download  # noqa
-from ._upload import upload  # noqa
+
+
+logger = logging.getLogger(__name__)
+
+
+class InvalidCredentialsError(Exception):
+    pass
+
+
+class StoreError(Exception):
+    """Base class for all storeapi exceptions.
+
+    :cvar fmt: A format string that daughter classes override
+    """
+
+    fmt = 'Daughter classes should redefine this'
+
+    def __init__(self, **kwargs):
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+    def __str__(self):
+        return self.fmt.format([], **self.__dict__)
+
+
+class SnapNotFoundError(StoreError):
+
+    fmt = 'The  "{name}" for {arch} was not found in {channel}.'
```

Comment:
Extra space fixed.
There is a test (with a bug in its name, fixed:  snapcraft.tests.store_tests.test_download.TestDownload_store.test_download_not_found ;) but that string is not exposed to the user as the exception is caught and re-raised as RuntimeError in _store as seems to be the practice in snapcraft.